python-checks/src/main/java/org/sonar/python/checks/CaughtExceptionsCheck.java

```diff
@@ -64,10 +73,43 @@ private static void checkExceptClause(SubscriptionContext ctx) {
       if (!canBeOrExtendBaseException(expression.type()) ||
         ((expression instanceof HasSymbol) && !inheritsFromBaseException(((HasSymbol) expression).symbol()))) {
 
-        ctx.addIssue(expression, MESSAGE);
+        var issue = (IssueWithQuickFix) ctx.addIssue(expression, MESSAGE);
+        if (exception instanceof HasSymbol) {
+          addQuickFix(issue, (HasSymbol) expression);
+        }
       }
     });
   }
+  
+  private static void addQuickFix(IssueWithQuickFix issue, HasSymbol expression) {
+    Optional.of(expression)
+      .map(HasSymbol::symbol)
+      .map(Symbol::usages)
+      .stream()
+      .flatMap(Collection::stream)
+      .filter(Usage::isBindingUsage)
+      .findFirst()
+      .map(Usage::tree)
+      .map(Tree::parent)
+      .filter(ClassDef.class::isInstance)
+      .map(ClassDef.class::cast)
+      .ifPresent(classDef -> {
```

Comment:
Somehow I have a feeling this could be simpler. I also have the feeling we are searching a second time for a class definition when it has probably been done when evaluating `(expression instanceof HasSymbol) && !inheritsFromBaseException(((HasSymbol) expression).symbol()))` of line 74. 

Perhaps we could reuse that a bit (e.g when we determined that the `classSymbol.canBeOrExtend(BASE_EXCEPTION)` is false, then we search for the definition of that symbol and create the quick fix from there).

Saying this, I'm really not sure making such change would lead to better code, happy to discuss it!